flexget/components/sites/sites/ncore.py

```diff
@@ -0,0 +1,130 @@
+
+logger = logger.bind(name='nCore')
+
+URL = 'https://ncore.pro'
+
+HEADERS = {'Content-type': 'application/x-www-form-urlencoded'}
+
+PASSKEY = ''
+
+CATEGORIES = {
+    "all",
+    "xvid_hun",
+    "xvid",
+    "dvd_hun",
+    "dvd",
+    "dvd9_hun",
+    "dvd9",
+    "hd_hun",
+    "hd",
+    "vidser_hun",
+    "xvidser",
+    "dvdser_hun",
+    "dvdser",
+    "hdser_hun",
+    "hdser",
+}
+
+SORT = {
+    'default': 'fid',
+    'date': 'fid',
+    'size': 'size',
+    'seeds': 'seeders',
+    'leechers': 'leechers',
+    'downloads': 'times_completed',
+}
+
+
+class UrlRewriteNcore:
+    """nCore urlrewriter."""
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'category': {
+                'type': 'array',
+                'items': {'type': 'string', 'enum': list(CATEGORIES)},
+                'default': [],
+            },
+            'sort_by': {'type': 'string', 'default': 'default', 'enum': list(SORT)},
+            'sort_reverse': {'type': 'boolean', 'default': False},
+        },
+        'required': ['username', 'password'],
+        'additionalProperties': False,
+    }
+
+    request_limiter = TokenBucketLimiter('ncore.pro/', 100, '5 seconds')
+
+    @plugin.internet(logger)
+    def search(self, task, entry, config):
+        """
+        Search for name from ncore.
+        """
+
+        data = {
+            "set_lang": "hu",
+            "submitted": "1",
+            "nev": config["username"],
+            "pass": config["password"],
+            "ne_leptessen_ki": 0,
+        }
+
+        page = task.requests.post(URL + "/login.php", data=data, headers=HEADERS)
+        soup = get_soup(page.content)
+        passkeyLine = str(soup.find('link', href=re.compile(r'rss\.php\?key=')))
+        PASSKEY = passkeyLine[passkeyLine.find("key=") : passkeyLine.find('"', 20, 90)]
+
+        entries = set()
+
+        for search_string in entry.get('search_strings', [entry['title']]):
+            data = {"mire": search_string, "miben": "name", "miszerint": SORT[config['sort_by']]}
+
+            if len(config['category']) > 0:
+                data["kivalasztott_tipus[]"] = config['category']
+                data["tipus"] = "kivalasztottak_kozott"
+            else:
+                data["tipus"] = "all_own"
+
+            if config['sort_reverse']:
+                data["hogyan"] = "DESC"
+
+            page = task.requests.post(URL + "/torrents.php", data=data, headers=HEADERS)
+            soup = get_soup(page.content)
+            for a in soup.findAll('a', title=re.compile(".+")):
+                if 'details' in a.get('href'):
+                    e = Entry()
+
+                    href = a.get('href')
+                    id = href[href.find("id=") + 3 :]
+
+                    e['title'] = a.text
+                    e['url'] = URL + '/torrents.php?action=download&id={0}&'.format(id) + PASSKEY
```

Comment:
f-strings are nice, using urllib is the best though

but this is probably fine for this PR if you don't feel like changing it